airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from unittest import TestCase
+
+from airbyte_cdk.test.entrypoint_wrapper import EntrypointOutput
+from airbyte_cdk.test.mock_http import HttpMocker
+from airbyte_cdk.test.mock_http.response_builder import (
+    FieldPath,
+    HttpResponseBuilder,
+    RecordBuilder,
+    create_record_builder,
+    create_response_builder,
+    find_template,
+)
+from airbyte_protocol.models import SyncMode
+
+from .config import ACCOUNT_ID, ConfigBuilder
+from .request_builder import get_account_request, get_ad_sets_request, get_ads_request, get_campaigns_request
+from .response_builder import get_account_response
+from .utils import config, read_output
+
+
+def _stream_record(stream: str, cursor_field: str = "updated_time") -> RecordBuilder:
+    return create_record_builder(
+        response_template=find_template(stream, __file__),
+        records_path=FieldPath("data"),
+        record_cursor_path=FieldPath(cursor_field),
+    )
+
+
+def _stream_response(stream: str) -> HttpResponseBuilder:
+    return create_response_builder(
+        response_template=find_template(stream, __file__),
+        records_path=FieldPath("data"),
+    )
+
+
+class TestIncludeDeleted(TestCase):
+    account_id = ACCOUNT_ID
+    filter_statuses_flag = "filter_statuses"
+    statuses = ["ACTIVE", "ARCHIVED"]
+
+    @staticmethod
+    def _read(config_: ConfigBuilder, stream_name: str, expecting_exception: bool = False) -> EntrypointOutput:
+        return read_output(
+            config_builder=config_,
+            stream_name=stream_name,
+            sync_mode=SyncMode.incremental,
+            expecting_exception=expecting_exception,
+        )
+
+    @HttpMocker()
+    def test_ads_stream(self, http_mocker: HttpMocker):
+        http_mocker.get(
+            get_account_request().build(),
+            get_account_response(),
+        )
+        filters = (
+            '[{"field":"ad.effective_status","operator":"IN","value":["ACTIVE","ARCHIVED"]},'
```

Comment:
Should we use `self. statuses` here to clarify the link between the config and the API call? The same comment applies for the other tests

Reply:
updated with self.statuses